api/lib/domain/usecases/improve-competence-evaluation.js

```diff
@@ -1,7 +1,22 @@
 const Assessment = require('../models/Assessment');
+const { MAX_REACHABLE_LEVEL } = require('../constants');
+const { ImproveCompetenceEvaluationForbiddenError } = require('../errors');
+
+module.exports = async function improveCompetenceEvaluation({
+  competenceEvaluationRepository,
+  getCompetenceLevel,
+  assessmentRepository,
+  userId,
+  competenceId,
+  domainTransaction
+}) {
+  const competenceEvaluation = await competenceEvaluationRepository.getByCompetenceIdAndUserId({ userId, competenceId });
+  const competenceLevel = await getCompetenceLevel({ userId, competenceId });
+
+  if (competenceLevel === MAX_REACHABLE_LEVEL) {
```

Comment:
Si par accident (ou par évolution des règles…) le niveau actuel est supérieur au niveau max autorisé on sera autorisé à progresser… J'aurais mis un `>=` 🤔 
```suggestion
  if (competenceLevel >= MAX_REACHABLE_LEVEL) {
```